### PR TITLE
Add base cli setup and testing. Close #27.

### LINKED ISSUE #27: Create scaffolding for command line interaction with server
## Specify type:
- Enhancement
___

## Description:
Provide python module and base CLI command class. 

Base class should provide:

- help setup
- validation
- command execution
- core processing for executing rest request and delegation of response handling

Module should also support general help and version information from command line

### PR DESCRIPTION
**Pull request by chambridge**

### CODE REVIEW

File: quipucords/cli/cli.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+from argparse import ArgumentParser
+import cli.auth as auth
+from cli.auth.add import AuthAddCommand
+from . import __version__
+
+
+# pylint: disable=too-few-public-methods
+class CLI(object):
+    """Class responsible for displaying ussage or matching inputs
+    to the valid set of commands supported by rho.
```
Comment (noahl) [inline quipucords/cli/cli.py]: Nit: "quipucords" instead of "rho"

Comment (noahl) [reply quipucords/cli/cli.py]: Or probably "qpc"

File: quipucords/cli/clicommand.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+# pylint: disable=too-few-public-methods
+class CliCommand(object):
+
+    """ Base class for all sub-commands. """
+    def __init__(self, subcommand, action):
+        self.subcommand = subcommand
+        self.action = action
+        self.args = None
+
+    def _validate_args(self):
```
Comment (noahl) [inline quipucords/cli/clicommand.py]: Nit: can we please remove the leading `_` from this and `do_command`? Those always bothered me in Rho.

File: quipucords/cli/tests.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+import contextlib
+import unittest
+import sys
+import six
+from cli.cli import CLI
+
+
+# pylint: disable=too-few-public-methods
+class HushUpStderr(object):
```
Comment (noahl) [inline quipucords/cli/tests.py]: Let's move `HushUpStderr` and `redirect_stdout` to a common test utilities library.

File: setup.py [hunk 4]
```diff
@@ -13,6 +23,12 @@
     ],
     include_package_data=True,
     license='GPLv3',
+    packages=find_packages(exclude=["test*.py"]),
     package_data={'': ['LICENSE']},
     url='https://github.com/quipucords/quipucords',
+    # non-python scripts go here
```
Comment (noahl) [inline setup.py]: This says non-python scripts go here, but `bin/qpc` is Python. No request for change, I just don't understand what this does.

File: Makefile [hunk 5]
```diff
@@ -31,16 +31,16 @@ install: build
...
 test-coverage:
-	coverage run --source=quipucords/ quipucords/manage.py test -v 2 quipucords;coverage report -m
+	coverage run --source=quipucords/ quipucords/manage.py test -v 2 quipucords/;coverage report -m
 
 lint-flake8:
 	flake8 . --ignore D203
 
 lint-pylint:
-	$(PYTHON) -m pylint --load-plugins=pylint_django quipucords/*/*.py
+	find . -name "*.py" | xargs $(PYTHON) -m pylint --load-plugins=pylint_django --disable=duplicate-code
```
Comment (noahl) [inline Makefile]: What code shows up when you enable the duplicate code checking? I would prefer to keep it enabled and spend a little more effort on refactoring if the warnings are at all reasonable.

File: quipucords/cli/cli.py [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+# version 3 (GPLv3). There is NO WARRANTY for this software, express or
+# implied, including the implied warranties of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv3
+# along with this software; if not, see
+# https://www.gnu.org/licenses/gpl-3.0.txt.
+#
+""" Quipucords Command Line Interface """
+
+from __future__ import print_function
+from argparse import ArgumentParser
```
Comment (noahl) [inline quipucords/cli/cli.py]: :+1: :+1: for changing to a non-deprecated argument parser!

File: quipucords/cli/clicommand.py (hunk 2 above)
Comment (chambridge) [reply quipucords/cli/clicommand.py]: is that not convention for denoting a method as private/internal? Meaning those two methods should not be called directly.

File: Makefile (hunk 5 above)
Comment (chambridge) [reply Makefile]: Only duplicate code is in the test files and I couldn't get the disable to work across files (several stackoverflows on this):

```
R:  1, 0: Similar lines in 2 files
==cli.auth.tests:21
==cli.tests:21
    def setUp(self):
        # Temporarily disable stderr for these tests, CLI errors clutter up
        # nosetests command.
        self.orig_stderr = sys.stderr
        sys.stderr = HushUpStderr()

    def tearDown(self):
        # Restore stderr
        sys.stderr = self.orig_stderr
 (duplicate-code)
R:  1, 0: Similar lines in 2 files
==cli.auth.tests:12
==cli.tests:12
import unittest
import sys
import six
from cli.cli import CLI
from cli.tests_utilities import HushUpStderr, redirect_stdout

 (duplicate-code)

```

File: quipucords/cli/cli.py (hunk 1 above)
Comment (chambridge) [reply quipucords/cli/cli.py]: updated

File: quipucords/cli/tests.py (hunk 3 above)
Comment (chambridge) [reply quipucords/cli/tests.py]: done

File: setup.py (hunk 4 above)
Comment (chambridge) [reply setup.py]: fixed. copy & paste from rho. I don't understand it either.

File: Makefile (hunk 5 above)
Comment (noahl) [reply Makefile]: Okay, fine with me.

File: quipucords/cli/clicommand.py (hunk 2 above)
Comment (noahl) [reply quipucords/cli/clicommand.py]: Yes, that's the convention. Maybe it bothers me because it's meant to be overridden, I'm not sure. Anyway, LGTM.

**Pull request merged**